modules/configuring-a-workload-to-use-shared-cpus.adoc

```diff
@@ -0,0 +1,132 @@
+
+//* You enabled the `TechPreviewNoUpgrade` feature set on the cluster. For more information about enabling a feature set, see the _Additional resources_ section.
+// +
+// [WARNING]
+// ====
+// Enabling the `TechPreviewNoUpgrade` feature set cannot be undone and prevents minor version updates. These feature sets are not recommended on production clusters.
+// ====
+
+.Procedure
+
+. Create a namespace by running the following command.
++
+[source,terminal]
+----
+$ oc create namespace <namespace_name>
+----
++
+[NOTE]
+====
+To use shared CPUs, you must use a namespace with the `workload.mixedcpus.openshift.io/allowed` annotation.
+====
+
+. Add the required annotation to the namespace by running the following command:
++
+[source,terminal]
+----
+$ oc annotate namespace <namespace_name> workload.mixedcpus.openshift.io/allowed=''
+----
++
+.Example output
+[source,terminal]
+----
+namespace/<namespace_name> annotate
+----
+
+. Create a `Pod` resource that requests shared CPUs:
+
+.. Create a YAML file that defines the `Pod` resource:
++
+--
+.Example `shared-cpu-pod.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata: 
+  name: dpdk-pod
+  metadata: <namespace_name> <1>
+spec:
+ runtimeClassName: performance-example-shared-cpu-pp <2>
+ containers:
+  - name: dpdk-cnt
+    image: image-address
+    resources:
+     requests: 
+      cpu: “2” <3>
+      memory: “100m” <4>
+      workload.openshift.io/enable-shared-cpus: ”1” <5>
+     limits:
+      cpu: “2”
+      memory: “100m”
+      workload.openshift.io/enable-shared-cpus: ”1”
+----
+<1> Specify the namespace with the `workload.mixedcpus.openshift.io/allowed=''` annotation.
+<2> Specify the performance profile with the shared CPU configuration in the format `performance-<performance_profile_name>`.
+<3> Specify the number of cores required.
++
+[IMPORTANT]
+====
+To use shared CPUs, you must create a pod with a QoS class of `Guaranteed`. To specify a pod with a QoS class of `Guaranteed`, the memory limit must equal the memory request, and the CPU limit must equal the CPU request. The CPU limit and CPU request values must be integers. 
+====
+<4> Specify the memory required. 
+<5> This is a boolean field to enable shared CPUs for the pod. Enter `1` to enable shared CPUs. The shared CPUs feature is disabled by default.
+--
+
+.. Create the `Pod` resource by running the following command:
++
+[source,terminal]
+----
+$ oc create -f shared-cpu-pod.yaml
+----
++
+.Example output
+[source,terminal]
+----
+pod/dpdk-pod created
+----
+
+. View the environment variable so you can pin lightweight tasks in your application to the shared CPU:
+
+.. Start an interactive shell within the target container by running the following command:
++
+[source,terminal]
+----
+$ oc exec -it dpdk-pod -- /bin/sh
+----
+
+.. View the environment variables within the container by running the following command:
++
+[source,terminal]
+----
+# printenv
```

Comment:
Perhaps grep to make the output shorter?

Reply:
Done